### ISSUE TITLE
[ChoiceGroup] Labels are truncated when text spacing is applied

### ISSUE THREAD

**Issue by srinidhiprabhu96** [microsoft/fluentui#21133]
### Environment Information

- **Package version(s)**: 8.29.2
- **Browser and OS versions**: Microsoft Edge 96.0.1054.62 (Official build) (64-bit), Windows 11

### Describe the issue:

When text spacing is applied, choice group labels are truncated. Text spacing is an [accessibility success criterion](https://www.w3.org/WAI/WCAG21/Understanding/text-spacing.html#:~:text=Understanding%20Success%20Criterion%201.4.12%3A%20Text%20Spacing%201%20Intent.,to%20read%20text.%203%20Examples.%20...%204%20Techniques) for users to improve their reading experience. 

To apply the text spacing style, use a browser extension called Stylus. Below are the steps to install and configure it –

1) Go to - https://chrome.google.com/webstore/detail/stylus/clngdbkpkpeebahjckkjfobafhncgmne?hl=en.
2) Click on ‘Add to Chrome’ button.
3) After adding, you should find Stylus icon on the tool bar, click it to open drop down menu.
4) Select ‘Manage Styles’ from the menu.
5) On the manage styles screen, click on ‘Create new Style’.
The New style screen would open, here give a suitable name for your new style, e.g. ‘text spacing’.
Add following CSS under ‘Code’ field (everything within the quotes)

"* {

line-height: 1.5 !important;

letter-spacing: 0.12em !important;

word-spacing: 0.16em !important;

}

p {

margin-bottom: 2em !important;

}"

Click on ‘Save’.
Now you should be able to see the style applied automatically on the page including the content displayed within iFrames. The Add on works both on Edge (Chromium) and Chrome

To remove the text spacing styling, go to the Stylus icon, the Stylus dialog would appear and check the ‘Remove all styles’ checkbox. The style could be reapplied with same checkbox again.

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/srinidhiprabhu96/pen/bGoaXpw

#### Actual behavior:

![image](https://user-images.githubusercontent.com/16191789/147614179-fc95e7da-0cc9-41a7-a83f-a30590f5343b.png)

#### Expected behavior:

Text should not be truncated

### Documentation describing expected behavior

https://www.w3.org/WAI/WCAG21/Understanding/text-spacing.html#:~:text=Understanding%20Success%20Criterion%201.4.12%3A%20Text%20Spacing%201%20Intent.,to%20read%20text.%203%20Examples.%20...%204%20Techniques

**Comment by gouttierre** [comment]
@srinidhiprabhu96 - Thanks for filing this issue with us. To set expectations, the developers will review this issue once capacity allows as we’re currently focused on larger coherence work items.

@TristanWatanabe - Would you be able to confirm if this is a regression, or if this behavior is an issue? 
cc @smhigley

**Comment by TristanWatanabe** [comment]
@srinidhiprabhu96 This is definitely an issue - the root cause is that the label's wrapper has a fixed `height` of 32px (code [here](https://github.com/microsoft/fluentui/blob/240b1c16b5493d3d2497af761f9c179be992378f/packages/react/src/components/ChoiceGroup/ChoiceGroupOption/ChoiceGroupOption.styles.ts#L19)) so it doesn't dynamically resize when text spacing (i.e. to the label) is applied.

**Comment by Neno83** [comment]
Hi Tristan, try onRenderLabel, it worked for me. 

https://developer.microsoft.com/en-us/fluentui#/controls/web/choicegroup

![image](https://user-images.githubusercontent.com/53298768/163115939-ba1e7e82-02fe-4de7-91bf-ae9336eb218e.png)

**Comment by smhigley** [comment]
This issue has come up before; the essential conflict is that the image/icon ChoiceGroup variants are constrained by their very structure. I've chatted with design about this, and the main takeaway was that it's not possible to resolve this specific design with flexible content. The only way to use this design is with _extremely_ short label text (read: one short word). Instead, we recommend people use the standard ChoiceGroup layout, with inline images or icons in the majority of cases.

This is why I added the following warning to both icon and image examples in our docs site:
> Warning: this ChoiceGroup option layout restricts the space that label text has to expand and wrap. This can cause issues for accessibility, when zoom and text spacing increase the length of words and lines, and also internationalization, since translated text length will vary.
> 
> We recommend using the standard layout with an inline icon for labels longer than a single short word.

Unfortunately there's not much else we can do for the moment. V9 won't have this issue (or this layout), so I'd mostly suggest just moving away from this design. Feel free to re-open the issue if this doesn't address your concerns.
